src/main/java/org/folio/rest/helper/GroupFiscalYearSummariesHelper.java

```diff
@@ -0,0 +1,106 @@
+package org.folio.rest.helper;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.mapping;
+import static java.util.stream.Collectors.reducing;
+import static java.util.stream.Collectors.toList;
+
+import io.vertx.core.Context;
+import org.folio.rest.jaxrs.model.Budget;
+import org.folio.rest.jaxrs.model.GroupFiscalYearSummary;
+import org.folio.rest.jaxrs.model.GroupFiscalYearSummaryCollection;
+import org.folio.rest.jaxrs.model.GroupFundFiscalYear;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+
+
+public class GroupFiscalYearSummariesHelper extends AbstractHelper {
+
+  private BudgetsHelper budgetsHelper;
+  private GroupFundFiscalYearHelper groupFundFiscalYearHelper;
+
+  public GroupFiscalYearSummariesHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
+    super(okapiHeaders, ctx, lang);
+    budgetsHelper = new BudgetsHelper(httpClient, okapiHeaders, ctx, lang);
+    groupFundFiscalYearHelper = new GroupFundFiscalYearHelper(httpClient, okapiHeaders, ctx, lang);
+  }
+
+  public CompletableFuture<GroupFiscalYearSummaryCollection> getGroupFiscalYearSummaries(int limit, int offset, String query) {
+    return budgetsHelper.getBudgets(limit, offset, query)
```

Comment:
It is better to use the max limit and offset=0 to search for budgets, because a chunk of budgets does not necessarily correspond to the chunk of groupFundFys.